src/pharmpy/modeling/estimation_steps.py

```diff
@@ -400,3 +400,108 @@ def set_evaluation_step(model: Model, idx: int = -1):
         newsteps = steps[0:-1] + newstep
     model = model.replace(estimation_steps=newsteps)
     return model.update_source()
+
+
+def add_predictions_residuals(model: Model, pred: List[str] = [], res: List[str] = []):
+    """Add predictions and/or residuals
+
+    Adds predictions and/or residuals to estimation step.
+
+    Parameters
+    ----------
+    model : Model
+        Pharmpy model
+    pred : list
+        List of predictions (e.g. ['IPRED', 'PRED'])
+    res : list
+        List of residuals (e.g. ['CWRES'])
+
+    Returns
+    -------
+    Model
+        Pharmpy model object
+
+    Examples
+    --------
+    >>> from pharmpy.modeling import *
+    >>> model = load_example_model("pheno")
+    >>> model.estimation_steps[-1].predictions
+    ('IPRED', 'PRED')
+    >>> model = add_predictions_residuals(model, pred=['CIPREDI'])
+    >>> model.estimation_steps[-1].predictions
+    ('CIPREDI', 'IPRED', 'PRED')
+
+    See also
+    --------
+    remove_predictions_residuals
+    set_estimation_step
+    add_estimation_step
+    remove_estimation_step
+    append_estimation_step_options
+    add_parameter_uncertainty_step
+    remove_parameter_uncertainty_step
+    """
+    steps = model.estimation_steps
+    old_predictions = steps[-1].predictions
+    new_predictions = tuple(sorted(set(old_predictions) | set(pred)))
+
+    old_residuals = steps[-1].residuals
+    new_residuals = tuple(sorted(set(old_residuals) | set(res)))
+    newstep = steps[-1].replace(predictions=new_predictions, residuals=new_residuals)
+
+    newsteps = steps[0:-1] + newstep
+    model = model.replace(estimation_steps=newsteps)
+    return model.update_source()
+
+
+def remove_predictions_residuals(model: Model, type: Literal['all', 'pred', 'res'] = 'all'):
```

Comment:
Also this should be split into `remove_predictions` and `remove_residuals`. Argument should be a list of what to remove or default to remove all (signalled by `None` as the default value perhaps)